ucsm_apis/admin/ldap.py

```diff
@@ -0,0 +1,875 @@
+    return mo
+
+
+def ldap_provider_group_exists(handle, name, **kwargs):
+    """
+    checks if ldap provider group exists
+
+    Args:
+        handle (UcsHandle)
+        name (string): name
+        **kwargs: key-value pair of managed object(MO) property and value, Use
+                  'print(ucscoreutils.get_meta_info(<classid>).config_props)'
+                  to get all configurable properties of class
+
+    Returns:
+        (True/False, MO/None)
+
+    Example:
+        ldap_provider_group_exists(handle, name="test_ldap_group_map")
+    """
+    try:
+        mo = ldap_provider_group_get(handle, name,
+                                     caller="ldap_provider_group_exists")
+    except UcsOperationError:
+        return (False, None)
+    mo_exists = mo.check_prop_match(**kwargs)
+    return (mo_exists, mo if mo_exists else None)
+
+
+def ldap_provider_group_delete(handle, name):
+    """
+    deletes ldap provider group
+
+    Args:
+        handle (UcsHandle)
+        name (string): name
+        descr (string): descr
+
+    Returns:
+        None
+
+    Raises:
+        UcsOperationError: If AaaProviderGroup is not present
+
+    Example:
+        ldap_provider_group_delete(handle, name="test_ldap_group_map")
+    """
+
+    mo = ldap_provider_group_get(handle, name,
+                                 caller="ldap_provider_group_delete")
+    handle.remove_mo(mo)
+    handle.commit()
+
+
+def ldap_provider_group_provider_add(handle, group_name, name,
+                                     order="lowest-available",
+                                     descr=None, **kwargs):
+    """
+    adds provider to ldap provider group
+
+    Args:
+        handle (UcsHandle)
+        group_name (string): ldap provider group name
+        name (string): name
+        order (string): order
+        descr (string): descr
+        **kwargs: Any additional key-value pair of managed object(MO)'s
+                  property and value, which are not part of regular args.
+                  This should be used for future version compatibility.
+    Returns:
+        AaaProviderRef : Managed Object
+
+    Raises:
+        UcsOperationError: If AaaProviderGroup or AaaProvider is not present
+
+    Example:
+        ldap_provider_group_provider_add(handle,
+                                        group_name="test_ldap_provider_group",
+                                        name="test_provider",
+                                        order="1")
+    """
+
+    from ucsmsdk.mometa.aaa.AaaProviderRef import AaaProviderRef
+
+    group_dn = _base_dn + "/ldap-ext/providergroup-" + group_name
+    group_mo = handle.query_dn(group_dn)
+    if not group_mo:
+        raise UcsOperationError("ldap_provider_group_provider_add",
+                                 "Ldap Provider Group does not exist.")
+
+    provider_dn = _base_dn + "/ldap-ext/provider-" + name
+    provider_mo = handle.query_dn(provider_dn)
+    if not provider_mo:
```

Comment:
Here we don't use provider_mo directly. Another mo `AaaProviderRef` is used to create link between `provider group` and `provider`. Above statement is just a check to make sure if `provider` exists.